common/rpc/interceptor/logtags/history_service_server_gen.go

```diff
@@ -0,0 +1,299 @@
+		return []tag.Tag{
+			tag.WorkflowID(r.GetExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetExecution().GetRunId()),
+		}
+	case *historyservice.ListQueuesRequest:
+		return nil
+	case *historyservice.ListTasksRequest:
+		return nil
+	case *historyservice.MergeDLQMessagesRequest:
+		return nil
+	case *historyservice.PollMutableStateRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetExecution().GetRunId()),
+		}
+	case *historyservice.PollWorkflowExecutionUpdateRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetRequest().GetUpdateRef().GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetRequest().GetUpdateRef().GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.PurgeDLQMessagesRequest:
+		return nil
+	case *historyservice.QueryWorkflowRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetRequest().GetExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetRequest().GetExecution().GetRunId()),
+		}
+	case *historyservice.ReapplyEventsRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetRequest().GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetRequest().GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.RebuildMutableStateRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetExecution().GetRunId()),
+		}
+	case *historyservice.RecordActivityTaskHeartbeatRequest:
+		return wt.fromTaskToken(r.GetHeartbeatRequest().GetTaskToken())
+	case *historyservice.RecordActivityTaskStartedRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.RecordChildExecutionCompletedRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetParentExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetParentExecution().GetRunId()),
+		}
+	case *historyservice.RecordWorkflowTaskStartedRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.RefreshWorkflowTasksRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetRequest().GetExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetRequest().GetExecution().GetRunId()),
+		}
+	case *historyservice.RemoveSignalMutableStateRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.RemoveTaskRequest:
+		return nil
+	case *historyservice.ReplicateEventsV2Request:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetWorkflowExecution().GetWorkflowId()),
+			tag.WorkflowRunID(r.GetWorkflowExecution().GetRunId()),
+		}
+	case *historyservice.ReplicateWorkflowStateRequest:
+		return []tag.Tag{
+			tag.WorkflowID(r.GetWorkflowState().GetExecutionInfo().GetWorkflowId()),
```

Comment:
does this have runid available as `r.GetWorkflowState().GetExecutionState().GetRunId()`?

Reply:
I had to slightly fix the approach to make it work.